### PR TITLE
Refactor documentation and code

### PR DESCRIPTION
**Pull request by mnojek**
- Updated all the links to documentation with `stable` version instead of `latest`.
- Updated links to our RoboCon talk
- Replaced redundant `visit_*` methods with shorthand versions
- Reformatted everything with `black`
- Improved some comments and prints
- Fixed some typos
- Added Python 3.10 as supported version (please confirm)

### CODE REVIEW

File: robocop/checkers/naming.py
Comment (bhirsz) [inline robocop/checkers/naming.py:711]: visit_KeywordCall and let's say visit_SuiteTeardown have different implementation - the first uses node.keyword and the other node.name. But that should be covered by me in the other PR (#576) already

Comment (mnojek) [reply robocop/checkers/naming.py:711]: I will merge this PR only after yours is merged. Then I will synchronize them.

File: robocop/checkers/errors.py
Comment (bhirsz) [inline robocop/checkers/errors.py:274]: Now when I look at this, I missed new RF 5.0 syntax here. visit_While, visit_Try. But it's still catched (ie in invalid-if). The question is if we need it at all - the good indicator would be running ``robocop --threshold E -r all`` on Robot Framework atests with this code and without visit_For, visit_If and compare number of reported issues.

File: robocop/checkers/lengths.py
Comment (bhirsz) [inline robocop/checkers/lengths.py:326]: Is there maybe general visit_Section? VisitorChecker is ModelVisitor so it there is chance it matched groups too.

File: setup.py
Comment (mnojek) [inline setup.py:18]: Is there anything that we do not support here? Can we put Python 3.10 to setup.py?

File: robocop/checkers/lengths.py
Comment (mnojek) [reply robocop/checkers/lengths.py:326]: I will look into it

File: robocop/checkers/lengths.py
Comment (bhirsz) [inline robocop/checkers/lengths.py:329]: Now when I see this checker I realize we need to update it to work with new RETURN statement (can you create issue?).

File: robocop/checkers/lengths.py
Comment (bhirsz) [inline robocop/checkers/lengths.py:337]: Check if it works with visit_While, visit_Try and also if we need to visit_For/visit_ForLoop at all

File: robocop/config.py
Comment (bhirsz) [inline robocop/config.py:80]: It wasn't clear error message? :D And it shouldn't print - maybe you could reraise it as ConfigGeneralError or ValueError so it will stop the execution

File: setup.py
Comment (bhirsz) [reply setup.py:18]: We should include 3.10 in our tests:
https://github.com/MarketSquare/robotframework-robocop/blob/ae27292ba5cb0972dd983135f14ed7b7187df0af/.github/workflows/unit-tests.yml#L1

Previously we actually had issue with 3.9 due to some syntax I've used so it's good idea to have it in checks

File: robocop/checkers/errors.py
Comment (bhirsz) [reply robocop/checkers/errors.py:274]: I've confirmed that we need it and we also need visit_While (we're missing some issues). I will add it & update tests for parsing-error

File: robocop/checkers/lengths.py
Comment (bhirsz) [reply robocop/checkers/lengths.py:337]: No need it for them - I will remove it.

File: robocop/checkers/lengths.py
Comment (bhirsz) [reply robocop/checkers/lengths.py:329]: I will do it.

**Review by bhirsz — APPROVED** [review]

File: robocop/checkers/naming.py
Comment (mnojek) [reply robocop/checkers/naming.py]: Updated

File: setup.py
Comment (mnojek) [reply setup.py:18]: Updated. We'll see the results soon.

File: robocop/checkers/lengths.py
Comment (mnojek) [reply robocop/checkers/lengths.py]: Updated with `visit_Section` (https://robot-framework.readthedocs.io/en/latest/autodoc/robot.parsing.model.html#robot.parsing.model.blocks.Section)

File: robocop/config.py
Comment (mnojek) [reply robocop/config.py:80]: Updated with `ConfigGeneralError`

File: setup.py
Comment (bhirsz) [reply setup.py:18]: Failed because 3.10 for yaml is 3.1 ;) Updated the code with '3.10'

**Review by bhirsz — APPROVED** [review]

**Pull request merged**
